daemon/src/projection.rs

```diff
@@ -444,6 +458,18 @@ impl Cfd {
                 complete_fee,
             } => {
                 self.expiry_timestamp = dlc.as_ref().map(|dlc| dlc.settlement_event_id.timestamp());
+
+                if let Some(dlc) = &dlc {
+                    let liquidation_price = match (self.role, self.position) {
+                        (Role::Maker, Position::Long) => *dlc.maker_liquidation_interval().end(),
```

Comment:
Can you explain how `.end` , `.start` work in combination with the function used inside of `maker_liquidation_interval`: `.min_by(|x, y|` 
I'm not sure I get it. 

My understanding is that `.min_by(|x, y|` returns you the smallest value in the list. This will be a range, e.g. `0..10`, i.e. the values where `maker` receives `0`. For getting the `long` liquidation price you can then do `end()`, i.e. you get `10` but for his `short` liquidation price you would need to get `.max_by..start` no?

Reply:
Here's my explanation with inline comments:

```rust
// cfd.rs

pub fn maker_liquidation_interval(&self) -> RangeInclusive<u64> {
    // Find CETs at settlement event. We use the settlement event
    // because all CFDs are guaranteed to have a settlement event, but
    // they might not have any liquidation events
    let (_, settlement_cets) = self
        .cets
        .iter()
        .find(|(id, _)| id == &&self.settlement_event_id)
        .expect("to have settlement CETs");

    // We search for the settlement CET that gives the maker the
    // least. That one must be the one that liquidates the maker and
    // will allow us to see the maker's liquidation price interval
    let liquidation_cet = settlement_cets
        .iter()
        .min_by(|x, y| x.maker_amount.cmp(&y.maker_amount))
        .expect("to have cets");

    liquidation_cet.range.clone()
}

// projection.rs

// Let's consider the maker side only, because the taker is equivalent
let liquidation_price = match (self.role, self.position) {
    // Long liquidation intervals go from 0-`end`, therefore `end`
    (Role::Maker, Position::Long) => *dlc.maker_liquidation_interval().end(),
    // Long liquidation intervals go from `start`-`max`, therefore `start`
    (Role::Maker, Position::Short) => *dlc.maker_liquidation_interval().start(),
    // ...
};
```

> My understanding is that .min_by(|x, y| returns you the smallest value in the list.

Continuing with the maker side: when we use `min_by` we are looking for the `Cet` that has the lowest `maker_amount`. That is, the CET that gives the the maker the lowest payout for a given oracle event. It's important to distinguish that from the CET that has the lowest _price_ in an interval!

> This will be a range, e.g. 0..10, i.e. the values where maker receives 0. 

We do return a range, but we don't know for sure that the payout will be zero, which is why we use `min_by`.

> For getting the long liquidation price you can then do end(), i.e. you get 10

I think we agree up to this point.

> but for his short liquidation price you would need to get .max_by..start no?

Nope, because we are ordering the CETs by the _payout amount_. The method `min_by` simply identifies the CET that gives the maker the least amount of money. And then if we know that the maker is going short we know that the (short) liquidation price interval will be of the form `start..=max`, where `start` will be the liquidation price.

Reply:
I will try to make it clearer, because it can be confusing.

Reply:
What do you think about https://github.com/itchysats/itchysats/pull/2812/commits/ded16b5038d5a887f0aff0d60dd7e54a0b64d570 now?

Reply:
> Nope, because we are ordering the CETs by the payout amount. The method min_by simply identifies the CET that gives the maker the least amount of money. And then if we know that the maker is going short we know that the (short) liquidation price interval will be of the form start..=max, where start will be the liquidation price.

Ah, that makes sense. https://github.com/itchysats/itchysats/commit/ded16b5038d5a887f0aff0d60dd7e54a0b64d570 is very clean. Thanks